src/super_gradients/modules/qarepvgg_block.py

```diff
@@ -0,0 +1,297 @@
+
+        self.nonlinearity = activation_type(**activation_kwargs)
+        self.se = se_type(**se_kwargs)
+
+        self.branch_3x3 = nn.Sequential()
+        self.branch_3x3.add_module(
+            "conv",
+            nn.Conv2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=3,
+                stride=stride,
+                padding=dilation,
+                groups=groups,
+                bias=False,
+                dilation=dilation,
+            ),
+        )
+        self.branch_3x3.add_module("bn", nn.BatchNorm2d(num_features=out_channels))
+
+        self.branch_1x1 = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=1,
+            stride=stride,
+            padding=0,
+            groups=groups,
+            bias=use_1x1_bias,
+        )
+
+        if use_residual_connection:
+            assert out_channels == in_channels and stride == 1
+
+            self.identity = Residual()
+
+            input_dim = self.in_channels // self.groups
+            self.id_tensor = torch.zeros((self.in_channels, input_dim, 3, 3))
+            for i in range(self.in_channels):
+                self.id_tensor[i, i % input_dim, 1, 1] = 1.0
+            self.id_tensor = self.id_tensor.to(dtype=self.branch_1x1.weight.dtype, device=self.branch_1x1.weight.device)
+        else:
+            self.identity = None
+
+        if use_alpha:
+            self.alpha = torch.nn.Parameter(torch.tensor([1.0]), requires_grad=True)
+        else:
+            self.alpha = 1.0
+
+        if self.use_post_bn:
+            self.post_bn = nn.BatchNorm2d(num_features=out_channels)
+        else:
+            self.post_bn = nn.Identity()
+
+        # placeholder to correctly register parameters
+        self.rbr_reparam = nn.Conv2d(
+            in_channels=self.branch_3x3.conv.in_channels,
+            out_channels=self.branch_3x3.conv.out_channels,
+            kernel_size=self.branch_3x3.conv.kernel_size,
+            stride=self.branch_3x3.conv.stride,
+            padding=self.branch_3x3.conv.padding,
+            dilation=self.branch_3x3.conv.dilation,
+            groups=self.branch_3x3.conv.groups,
+            bias=True,
+        )
+
+        self.partially_fused = False
+        self.fully_fused = False
+
+        if not build_residual_branches:
+            self.fuse_block_residual_branches()
+
+    def forward(self, inputs):
+        if self.fully_fused:
+            return self.se(self.nonlinearity(self.rbr_reparam(inputs)))
+
+        if self.partially_fused:
+            return self.se(self.nonlinearity(self.post_bn(self.rbr_reparam(inputs))))
+
+        if self.identity is None:
+            id_out = 0.0
+        else:
+            id_out = self.identity(inputs)
+
+        return self.se(self.nonlinearity(self.post_bn(self.branch_3x3(inputs) + self.alpha * self.branch_1x1(inputs) + id_out)))
```

Comment:
can please break it into a few lines? 
later when it crashes for some reason, it will be very hard to know which one f these 5 layers caused the crash